runtime/lua/vim/loader.lua

```diff
@@ -170,20 +192,6 @@ function Loader.loader_lib(modname)
   return '\ncache_loader_lib: module ' .. modname .. ' not found'
 end
 
---- `loadfile` using the cache
----@param filename? string
----@param mode? "b"|"t"|"bt"
----@param env? table
----@param hash? CacheHash
----@return function?, string?  error_message
----@private
--- luacheck: ignore 312
-function Loader.loadfile(filename, mode, env, hash)
```

Comment:
LuaJIT does support the mode and env parameters. See https://luajit.org/extensions.html

I'm actually using the `env` parameter in my personal configs to show virtual text where errors happened and with print statements when running a lua file.

The `env` parameter is useful when you want to sandbox or override the globals object when loading a lua module.

Removing this would break compatibility with LuaJIT, which is not something I think we want...

Reply:
Sounds good. I'll add it back and add a comment that luajit supports this and it is 5.1 compatible.